extensions/Worker.Extensions.CosmosDB/src/Config/CosmosDBExtensionOptions.cs

```diff
@@ -11,5 +11,10 @@ public class CosmosDBExtensionOptions
         /// Gets or sets the CosmosClientOptions.
         /// </summary>
         public CosmosClientOptions ClientOptions { get; set; } = new() { ConnectionMode = ConnectionMode.Gateway };
+
+        internal CosmosDBExtensionOptions Clone()
+        {
+            return (CosmosDBExtensionOptions)this.MemberwiseClone();
```

Comment:
Memberwise clone is a shallow copy. We will still have the same instance of `ClientOptions`. I am only concerned about how we apply `ApplicationPreferredRegions` to the client options, so no need for a clone here. Instead clone `ClientOptions` directly before applying preferred regions.

Reply:
Oh I understand what you're saying now, I went down a different rabbit hole 🕳️